src/base/path.cpp

```diff
@@ -181,6 +183,12 @@ void Path::removeExtension()
     m_pathStr.chop(extension().size());
 }
 
+void Path::removeExtension(const QString &ext)
```

Comment:
maybe combine them into one?
```c++
void Path::removeExtension(const QString &ext = {})
{
    Q_ASSERT(ext.startsWith(QLatin1Char('.')));

    if (ext.isEmpty())
    {
        m_pathStr.chop(extension().size());
        return;
    }

    if (hasExtension(ext))
        m_pathStr.chop(ext.size());
}
```

Reply:
> maybe combine them into one?

There are two completely different code paths so I would prefer they to be resolved in compile time.